Hacking.md

```diff
@@ -24,3 +24,128 @@ When P/Invoking a Swift function, you must keep in mind that Swift may pass some
+        00000000000039e2         mov        esi, 0x5
+        00000000000039e7         mov        edx, 0x1
+        00000000000039ec         call       imp___stubs__$sSS21_builtinStringLiteral17utf8CodeUnitCount7isASCIISSBp_BwBi1_tcfC ; Swift.String.init(_builtinStringLiteral: Builtin.RawPointer, utf8CodeUnitCount: Builtin.Word, isASCII: Builtin.Int1) -> Swift.String
+        00000000000039f1         mov        rdi, rax
+        00000000000039f4         mov        rsi, rdx
+        00000000000039f7         call       imp___stubs__$s7SwiftUI18LocalizedStringKeyV13stringLiteralACSS_tcfC ; SwiftUI.LocalizedStringKey.init(stringLiteral: Swift.String) -> SwiftUI.LocalizedStringKey
+        00000000000039fc         mov        qword [rbp+var_80], rax
+        0000000000003a00         mov        qword [rbp+var_88], rdx
+        0000000000003a07         mov        byte [rbp+var_89], cl
+        0000000000003a0d         mov        qword [rbp+var_98], r8
+        0000000000003a14         call       _$s7SwiftUI4TextV_9tableName6bundle7commentAcA18LocalizedStringKeyV_SSSgSo8NSBundleCSgs06StaticI0VSgtcfcfA0_ ; default argument 1 of SwiftUI.Text.init(_: SwiftUI.LocalizedStringKey, tableName: Swift.String?, bundle: __C.NSBundle?, comment: Swift.StaticString?) -> SwiftUI.Text
+        0000000000003a19         mov        qword [rbp+var_A0], rax
+        0000000000003a20         mov        qword [rbp+var_A8], rdx
+        0000000000003a27         call       _$s7SwiftUI4TextV_9tableName6bundle7commentAcA18LocalizedStringKeyV_SSSgSo8NSBundleCSgs06StaticI0VSgtcfcfA1_ ; default argument 2 of SwiftUI.Text.init(_: SwiftUI.LocalizedStringKey, tableName: Swift.String?, bundle: __C.NSBundle?, comment: Swift.StaticString?) -> SwiftUI.Text
+        0000000000003a2c         mov        qword [rbp+var_B0], rax
+        0000000000003a33         call       _$s7SwiftUI4TextV_9tableName6bundle7commentAcA18LocalizedStringKeyV_SSSgSo8NSBundleCSgs06StaticI0VSgtcfcfA2_ ; default argument 3 of SwiftUI.Text.init(_: SwiftUI.LocalizedStringKey, tableName: Swift.String?, bundle: __C.NSBundle?, comment: Swift.StaticString?) -> SwiftUI.Text
+
+- Now switch to the pseudo code view and uncheck "Remove potential dead code" and "Remove NOPs" at the top. This sometimes hides crucial information.
+- Now in the psuedo code scroll down until you see
+
+        testLib.SetViewBackground<A, B where A: SwiftUI.View, B: SwiftUI.View>
+
+- Notice the parameter order of `(&var_70, &var_78, rdx, rcx)`
+    * where var_70 is the result of the `SwiftUI.Text.init()` calls
+    * where var_78 is the result of the static call to the `SwiftUI.Color.red.getter`
+    * where rdx represents the `*type metadata for SwiftUI.Text`
+    * where rcs represents the `*type metadata for SwiftUI.Color`
+Also worth noting that...
+    * 8 holds the `*protocol witness table for SwiftUI.Text`
+    * r9 holds the `*protocol witness table for SwiftUI.Color`
+These are hidden parameters which we'll need later.
```

Comment:
Should we also perhaps link to the register usage doc in the Swift repo here?

Reply:
Yes good call!

Reply:
I've also added a link to the SwiftUI.tbd mangled API list.